ldms/src/ldmsd/ldmsd_request.c

```diff
@@ -6361,12 +6367,57 @@ static char * __set_stats_as_json(size_t *json_sz)
 		goto __APPEND_ERR;
 	s = buff;
 
+	ldmsd_cfg_lock(LDMSD_CFGOBJ_UPDTR);
+	for (updtr = ldmsd_updtr_first(); updtr;
+			updtr = ldmsd_updtr_next(updtr)) {
+		if (!LIST_EMPTY(&updtr->match_list)) {
+			LIST_FOREACH(match, &updtr->match_list, entry) {
+				ldmsd_prdcr_ref_t ref;
+				for (ref = ldmsd_updtr_prdcr_first(updtr); ref;
+						ref = ldmsd_updtr_prdcr_next(ref)) {
+					ldmsd_prdcr_lock(ref->prdcr);
+					ldmsd_prdcr_set_t prd_set;
+					for (prd_set = ldmsd_prdcr_set_first(ref->prdcr); prd_set;
+							prd_set = ldmsd_prdcr_set_next(prd_set)) {
+						rc = regexec(&match->regex, prd_set->inst_name, 0, NULL, 0);
+						if (rc)
+							continue;
+						freq = 1000000 / (double)prd_set->updt_hint.intrvl_us;
```

Comment:
`prd_set->updt_interval` is the current update interval. `prd_set->updt_hint.intervl_us` is the update hint received from the sampler daemon. I guess you want to calculate the current set load. `prd_set->updt_interval` is the value you want to use.

I think both values are the same because we usually set the sample interval and the update interval to 1 second.  The values will be different when the sample interval differs from the update interval.